mungegithub/github/github.go

```diff
@@ -630,14 +658,26 @@ func (config *Config) needsBranchProtection(branch *github.Branch, contexts []st
 
 // SetBranchProtection protects a branch and sets the required contexts
 func (config *Config) SetBranchProtection(name string, contexts []string) error {
-	branch, resp, err := config.client.Repositories.GetBranch(config.Org, config.Project, name)
-	config.analytics.GetBranch.Call(config, resp)
+	prot, resp, err := config.client.Repositories.GetBranchProtection(
+		context.Background(),
+		config.Org,
+		config.Project,
+		name,
+	)
+	config.analytics.GetBranchProtection.Call(config, resp)
 	if err != nil {
-		glog.Errorf("Got error getting branch %s: %v", name, err)
-		return err
+		b, errRead := ioutil.ReadAll(resp.Body)
```

Comment:
This whole error handling is just gross. We have to do all this because `GetBranchProtection` can't differentiate between and error and being unset. Blech blech blech.